src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -11436,18 +11436,61 @@ class C<T> {}
             // encode these in metadata, and may need to work with the runtime team to define a new format.
             comp.VerifyDiagnostics(
-                // (4,7): error CS8911: Using a function pointer type in a 'typeof' in an attribute is not supported.
+                // (4,7): error CS8911: Using a function pointer type in an attribute while in a 'default' of an enum or a 'typeof' is not supported.
                 // [Attr(typeof(delegate*<void>))]
                 Diagnostic(ErrorCode.ERR_FunctionPointerTypesInAttributeNotSupported, "typeof(delegate*<void>)").WithLocation(4, 7),
-                // (5,7): error CS8911: Using a function pointer type in a 'typeof' in an attribute is not supported.
+                // (5,7): error CS8911: Using a function pointer type in an attribute while in a 'default' of an enum or a 'typeof' is not supported.
                 // [Attr(typeof(delegate*<void>[]))]
                 Diagnostic(ErrorCode.ERR_FunctionPointerTypesInAttributeNotSupported, "typeof(delegate*<void>[])").WithLocation(5, 7),
-                // (6,7): error CS8911: Using a function pointer type in a 'typeof' in an attribute is not supported.
+                // (6,7): error CS8911: Using a function pointer type in an attribute while in a 'default' of an enum or a 'typeof' is not supported.
                 // [Attr(typeof(C<delegate*<void>[]>))]
                 Diagnostic(ErrorCode.ERR_FunctionPointerTypesInAttributeNotSupported, "typeof(C<delegate*<void>[]>)").WithLocation(6, 7)
             );
         }
 
+        [Fact, WorkItem(65594, "https://github.com/dotnet/roslyn/issues/65594")]
+        public void Attribute_Default_Enum()
+        {
+            var source = """
+                class A : System.Attribute
+                {
+                    public A(object o) { }
+                }
+
+                class B<T>
+                {
+                    public enum E { }
+                }
+
+                [A(default(B<delegate*<void>[]>.E))]
+                class C { }
+                """;
+
+            // https://github.com/dotnet/roslyn/issues/48765 tracks enabling support for this scenario.
+            CreateCompilation(source).VerifyDiagnostics(
+                // (11,4): error CS8911: Using a function pointer type in an attribute while in a 'default' of an enum or a 'typeof' is not supported.
+                // [A(default(B<delegate*<void>[]>.E))]
+                Diagnostic(ErrorCode.ERR_FunctionPointerTypesInAttributeNotSupported, "default(B<delegate*<void>[]>.E)").WithLocation(11, 4));
+        }
+
+        [Fact, WorkItem(65594, "https://github.com/dotnet/roslyn/issues/65594")]
+        public void Attribute_Default_GenericArgument()
+        {
+            var source = """
+                class A : System.Attribute
+                {
+                    public A(object o) { }
+                }
+
+                class B<T> { }
+
+                [A(default(B<delegate*<void>[]>))]
```

Comment:
>B<delegate*<void>[] [](http://example.com/codeflow?start=27&length=19)

Consider testing with function pointer type used at the top.

Reply:
Do you mean `[A(default(delegate*<void>))]`? I'm not sure how to do that, it's not possible to have attribute with argument of type `delegate*<void>`, nor is it possible to convert it to `object`, for example.

I could do just an array, though, i.e., `[A(default(delegate*<void>[]))]`.

Reply:
Let's demonstrate with a test that this cannot be done. The array case is probably interesting too, but not the scenarios I am asking for here.

Reply:
Alright, added a test.